cmd/lncli/types.go

```diff
@@ -66,3 +87,124 @@ func NewUtxoFromProto(utxo *lnrpc.Utxo) *Utxo {
 		Confirmations: utxo.Confirmations,
 	}
 }
+
+// ListPaymentsResponse contains a list of payments made by this node.
+type ListPaymentsResponse struct {
+	// Payments is the list of payments made.
+	Payments []Payment `json:"payments"`
+}
+
+// Payment encapsulates all information about a particular payment including all
+// HTLCs made in attempt to pay the recipient.
+type Payment struct {
```

Comment:
What more beyond this would've needed to have been added that inclined you towards the direction of forking the proto lib?

Reply:
Minimally `QueryRoutes` and build `BulidRoute` would have needed wrappers. Also the `Path` field on the `HTLCAttempt` will be replaced with `Route`, which itself will need a wrapper and then of course the internal `Hop` message, so four layers deep. Turned out to be way more code than it was worth for something that can be totally automated. Also helps to keep our `_types.go` files from expanding..